packages/teleport/src/config.ts

```diff
@@ -264,8 +267,11 @@ const cfg = {
     return generatePath(cfg.routes.appLauncher, { ...params });
   },
 
-  getPlayerRoute(params: UrlPlayerParams) {
-    return generatePath(cfg.routes.player, { ...params });
+  getPlayerRoute(params: UrlPlayerParams, search: UrlPlayerSearch) {
```

Comment:
why can't `recordingType` be part of `UrlPlayerParam`?

Reply:
[Related](https://github.com/gravitational/webapps/pull/572#discussion_r797713609)

It looks like it could be, as in I can do

```typescript
cfg.routes.player: '/web/cluster/:clusterId/session/:sid', // ?recordingType=ssh|desktop

export interface UrlPlayerParams {
  clusterId: string;
  sid: string;
  recordingType: RecordingType;
}

getPlayerRoute(params: UrlPlayerParams) {
    return generatePath(cfg.routes.player, { ...params }) + `?recordingType=${params.recordingType}`;
  },
```

and `generatePath` will just ignore the additional `recordingType` that it doesn't see in `cfg.routes.player`. 

However afaik throughout the rest of the code we are using `*Params` to mean "params` in the react router sense of the word, namely path parameters, and so I decided to make it clear that `recordingType` is not one of those, it's distinct (a query parameter, or "search" since that's the nomenclature used by [`URLSearchParams`](https://github.com/gravitational/webapps/blob/isaiah/desktop-session-recording/packages/teleport/src/services/history/history.ts#L115-L119))

Reply:
is `recordingType` really a query type though? to me query seems more like "search for something", but we aren't searching

Reply:
The lines are a bit blurry with a SPA I suppose. I was thinking of the path parameters as "distinct path to the backend resource" and the query parameters as "other". I originally had what you suggested, but changed it owing to [Zac's suggestion](https://github.com/gravitational/webapps/pull/572#discussion_r795799523).

Reply:
i see, i'm curious about @hatched opinions as well

Reply:
i also maybe think these should be two distinct routes

Reply:
My reasoning for suggesting the query param..

`/web/cluster/:clusterId/session/:sid/` is unique and refers to a single resource. The session is either a desktop session or an SSH session. It can never be both, meaning we could never have `/web/cluster/foo/session/123/desktop` and `/web/cluster/foo/session/123/ssh`.

The path should describe exactly **what** I want to see. `/web/cluster/:clusterId/session/:sid/` uniquely describes the session I want to view.

The query params are _paramaters_ to the query that can adjust **how** I want to see it. Other examples of using query params in this way are APIs that take a `?format=json` or `?sort=desc` param. Again, these aren't specifying what resource, but changing how it is presented or displayed.

Reply:
@zmb3 that makes sense, thanks for the explanation!